docker-compose.yml

```diff
@@ -33,73 +37,28 @@ services:
     depends_on:
       - db
       - middleman
+      - redis
     environment:
       - DB_HOSTNAME=db
       - DB_USERNAME=postgres
       - DB_PASSWORD=developmentpassword
       - CC_TEST_REPORTER_ID=${CC_TEST_REPORTER_ID}
       - AGENT_JOBSTATUS=${AGENT_JOBSTATUS}
       - GIT_BRANCH=${GIT_BRANCH}
+      - REDIS_URL=redis://redis:6379
 
-  bg-geocode:
-    build:
-      args:
-        - DOCKER_REPOSITORY=${dockerHubUsername:-dfedigital}
-        - TEACHER_TRAINING_API=${dockerHubImageName:-teacher-training-api}
-        - CODE_VERSION=${GIT_BRANCH:-latest}
-      context: bg-geocode
-      cache_from:
-        - ${dockerHubUsername:-dfedigital}/teacher-training-bg-geocode:${GIT_BRANCH:-latest}
-    image: ${dockerHubUsername:-dfedigital}/teacher-training-bg-geocode:${GIT_BRANCH:-latest}
-    volumes:
-      - .:/app
-    depends_on:
-      - web
-      - db
-    environment:
-      - DB_HOSTNAME=db
-      - DB_USERNAME=postgres
-      - DB_PASSWORD=developmentpassword
-      - SETTINGS__APPLICATION=teacher-training-api-bg-geocode
-
-  bg-mailer:
-    build:
-      args:
-        - DOCKER_REPOSITORY=${dockerHubUsername:-dfedigital}
-        - TEACHER_TRAINING_API=${dockerHubImageName:-teacher-training-api}
-        - CODE_VERSION=${GIT_BRANCH:-latest}
-      context: bg-mailer
-      cache_from:
-        - ${dockerHubUsername:-dfedigital}/teacher-training-bg-mailer:${GIT_BRANCH:-latest}
-    image: ${dockerHubUsername:-dfedigital}/teacher-training-bg-mailer:${GIT_BRANCH:-latest}
-    volumes:
-      - .:/app
-    depends_on:
-      - web
-      - db
-    environment:
-      - DB_HOSTNAME=db
-      - DB_USERNAME=postgres
-      - DB_PASSWORD=developmentpassword
-      - SETTINGS__APPLICATION=teacher-training-api-bg-mailer
-
-  bg-jobs:
-    build:
-      args:
-        - DOCKER_REPOSITORY=${dockerHubUsername:-dfedigital}
-        - TEACHER_TRAINING_API=${dockerHubImageName:-teacher-training-api}
-        - CODE_VERSION=${GIT_BRANCH:-latest}
-      context: bg-jobs
-      cache_from:
-        - ${dockerHubUsername:-dfedigital}/teacher-training-bg-jobs:${GIT_BRANCH:-latest}
-    image: ${dockerHubUsername:-dfedigital}/teacher-training-bg-jobs:${GIT_BRANCH:-latest}
-    volumes:
+  bg:
+    image: ${dockerHubUsername:-dfedigital}/teacher-training-api:${GIT_BRANCH:-latest}
+    command: bundle exec sidekiq -c 5 -C config/sidekiq.yml
```

Comment:
Just to note that the Dockerfiles for the bg containers were running `rails db:migrate` before `sidekiq`. I think it's ok here in docker-compose not to do that, but we should consider doing the `rails db:migrate` when we run these containers in the deployed environments.

Reply:
In docker compose there is a dependency between web and bgjobs so migrate will always have run when bg starts.

In deployed environments, it may be dangerous to add migrate to bgjobs as it may end up running in parallel webapp//bgjobs. We could simply add a similar dependency in terraform so the webapp is always updated first and runs migrate. So bgjobs wouldn't have to run it.

Reply:
[Rails migrations are safe to run concurrently](https://github.com/rails/rails/pull/22122). The problem you describe already happens in environments that have multiple web servers that can be spun-up at the same time. Putting `db:migrate` into the startup of bg-jobs just means that that process should always be safe to run (it won't error out saying that the db is the wrong version).

That said, using a dependency in terraform is even better, happy to go with that. :)